src/linear_algebra/linear_pmap.lean

```diff
@@ -0,0 +1,374 @@
+/-
+Copyright (c) 2020 Yury Kudryashov All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import linear_algebra.basic
+
+/-!
+# Partially defined linear maps
+
+A `linear_pmap R E F` is a linear map from a submodule of `E` to `F`. We define
+a `semilattice_inf_bot` instance on this this, and define three operations:
+
+* `mk_span_singleton` defines a partial linear map defined on the span of a singleton.
+* `sup` takes two partial linear maps `f`, `g` that agree on the intersection of their
+  domains, and returns the unique partial linear map on `f.domain ⊔ g.domain` that
+  extends both `f` and `g`.
+* `Sup` takes a `directed_on (≤)` set of partial linear maps, and returns the unique
+  partial linear map on the `Sup` of their domains that extends all these maps.
+-/
```

Comment:
Could you mention that their main use is the proof of Hahn-Banach and variations?